application_sdk/observability/utils.py

```diff
@@ -26,16 +26,13 @@ class WorkflowContext(BaseModel):
 
 
 def get_observability_dir() -> str:
-    """Build the observability path.
-
-    Args:
-        path: The path to build the observability path from.
+    """Build the observability path using deployment name.
 
     Returns:
-        str: The built observability path.
+        str: The built observability path using deployment name.
     """
     return os.path.join(
-        TEMPORARY_PATH, OBSERVABILITY_DIR.format(application_name=APPLICATION_NAME)
+        TEMPORARY_PATH, OBSERVABILITY_DIR.format(deployment_name=DEPLOYMENT_NAME)
```

Comment:
deployment name will be either `local` or `production` this will lead to overrwrites. from what I remember we discussed to review this further if deployment-name could work

```
artifacts/apps/{appName}/{deploymentName}/observability}
```